unlock-app/src/hooks/useProvider.ts

```diff
@@ -185,32 +185,35 @@ export const useProvider = (config: any) => {
       resetProvider(newProvider)
     } else {
       try {
-        const changeNetworkRequest = provider.send(
-          'wallet_switchEthereumChain',
-          [
-            {
-              chainId: `0x${id.toString(16)}`,
-            },
-          ],
-          account
-        )
-        await ToastHelper.promise(changeNetworkRequest, {
-          loading: `Changing network to ${name}. Please Approve on your wallet.`,
-          error: `Error in changing network to ${name}`,
-          success: `Successfully changed network to ${name}`,
-        })
-        setNetwork(id)
+        const changeNetworkRequest = provider
+          .send(
+            'wallet_switchEthereumChain',
+            [
+              {
+                chainId: `0x${id.toString(16)}`,
+              },
+            ],
+            account
+          )
+          .catch((switchError: any) => {
+            if (switchError.code === 4902 || switchError.code === -32603) {
+              return addNetworkToWallet(id)
+            } else {
+              throw switchError
+            }
+          })
+        try {
+          await ToastHelper.promise(changeNetworkRequest, {
+            loading: `Changing network to ${name}. Please approve in your wallet.`,
+            error: `We could not switch to ${name}. Try adding it manually in your wallet.`,
+            success: `Successfully changed network to ${name}.`,
+          })
+          setNetwork(id)
+        } catch (error) {
+          console.error(error)
+        }
       } catch (switchError: any) {
         // This error code indicates that the chain has not been added to the provider yet.
-        if (switchError.code === 4902) {
-          try {
-            await addNetworkToWallet(id)
-          } catch (addError) {
-            ToastHelper.error(
-              'Network could not be added. Please try manually adding it to your wallet'
-            )
-          }
-        }
       }
     }
```

Comment:
🤔 This catch is redundant even though it is catching the switch error from above. Want to add a log or refactor this to be a single try-catch block?